cmd/api-linter/cli.go

```diff
@@ -177,7 +181,7 @@ func (c *cli) lint(rules lint.RuleRegistry, configs lint.Configs) error {
 	}
 
 	// Create a linter to lint the file descriptors.
-	l := lint.New(rules, configs)
+	l := lint.New(rules, configs, lint.Debug(c.DebugFlag))
```

Comment:
nit: would a new parameter to New() be easier? In my general experience constructors just pass in the parameter that need to be configured, rather than builder options.

That said I guess this allows for one or more optional parameters, which I suppose Go doesn't allow you to do.

Reply:
oops.. I guess I merged the PR without giving you time to react. 

I'll resolve this and if you feel the refactor is beneficial, please add it!

Reply:
Any change to the function signature is technically a breaking change.

Appending a variadic argument is, but a "lesser" one in that existing usages don't break in place unless some "advanced" methods are being used (type aliasing the signature, passing it as a "function pointer", using reflect (idk why anyone would here)).

And as you said, this allows us to add stuff later that would otherwise constitute another "hard" breaking change.

Reply:
That said, if we want to avoid the break altogether, I'd recommend using a function to toggle it on like

```go
l := lint.New(rules, configs)
l.Debug(true)
```